File: app/src/main/java/ardjomand/leonardo/nutrimeal/data/AuthRepository.java
package ardjomand.leonardo.nutrimeal.data;

import ardjomand.leonardo.nutrimeal.auth.User;

public interface AuthRepository {

    interface Interactor {

        interface CreateAccountCallback {

            void onSuccess();

            void onFailure(Exception e);
        }

        void createAccount(CreateAccountCallback createAccountCallback, User user,
                           String password, String type);


        interface LogInCallback {

            void onSuccess();

            void onFailure(Exception e);
        }

        void logIn(LogInCallback logInCallback, String email, String password);


        interface GetCurrentUserCallback {

            void onUserLoggedIn(User user);

            void onUserLoggedOut();

            void onFailure(Exception e);
        }

        void getCurrentUser(GetCurrentUserCallback getCurrentUserCallback);

    }

    interface Presenter {

    }
}

File: app/src/main/java/ardjomand/leonardo/nutrimeal/data/AuthRepository.kt
package ardjomand.leonardo.nutrimeal.data

import ardjomand.leonardo.nutrimeal.data.pojos.User

interface AuthRepository {

    interface Interactor {

        interface CreateAccountCallback {

            fun onSuccess()

            fun onFailure(e: Exception)
        }

        fun createAccount(createAccountCallback: CreateAccountCallback,
                          user: User,
                          password: String,
                          type: String)


        interface LogInCallback {

            fun onSuccess()

            fun onFailure(e: Exception)
        }

        fun logIn(logInCallback: LogInCallback,
                  email: String,
                  password: String)


        interface GetCurrentUserCallback {

            fun onUserLoggedIn(user: User)

            fun onUserLoggedOut()

            fun onFailure(e: Exception)
        }

        fun getCurrentUser(getCurrentUserCallback: GetCurrentUserCallback)

    }

    interface Presenter
}
